contracts/contracts/strategies/FrxEthRedeemStrategy.sol

```diff
@@ -0,0 +1,250 @@
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+        safeApproveAllTokens();
+    }
+
+    /**
+     * @notice deposit() function not used for this strategy. Use depositAll() instead.
+     */
+    function deposit(address, uint256) public override onlyVault nonReentrant {
+        // This method no longer used by the VaultAdmin, and we don't want it
+        // to be used by VaultCore.
+        require(false, "use depositAll() instead");
+    }
+
+    /**
+     * @notice Takes all given frxETH and creates new redeem tickets
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 frxETHStart = frxETH.balanceOf(address(this));
+        require(frxETHStart > 0, "No frxETH to redeem");
+        uint256 frxETHRemaining = frxETHStart;
+
+        while (frxETHRemaining > 0) {
+            uint256 amount = frxETHRemaining > maxRedeemTicket
+                ? maxRedeemTicket
+                : frxETHRemaining;
+            uint256 nftId = redemptionQueue.enterRedemptionQueue(
+                address(this),
+                uint120(amount)
+            );
+            frxETHRemaining -= amount;
+            emit RedeemNFTMinted(nftId, amount);
+        }
+
+        require(
+            frxETH.balanceOf(address(this)) == 0,
+            "Not all FraxEth sent to redemption queue"
+        );
+        outstandingRedeems += frxETHStart; // Single set for gas reasons
+
+        // This strategy claims to support WETH, so it is posible for
+        // the vault to transfer WETH to it. This returns any deposited WETH
+        // to the vault so that it is not lost for balance tracking purposes.
+        uint256 wethBalance = weth.balanceOf(address(this));
+        if (wethBalance > 0) {
+            // slither-disable-next-line unchecked-transfer
+            weth.transfer(vaultAddress, wethBalance);
+        }
+
+        emit Deposit(address(frxETH), address(redemptionQueue), frxETHStart);
+    }
+
+    /**
+     * @notice Withdraw an asset from the underlying platform
+     * @param _recipient Address to receive withdrawn assets
+     * @param _asset Address of the asset to withdraw
+     * @param _amount Amount of assets to withdraw
+     */
+    function withdraw(
+        // solhint-disable-next-line no-unused-vars
+        address _recipient,
+        // solhint-disable-next-line no-unused-vars
+        address _asset,
+        // solhint-disable-next-line no-unused-vars
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        // Does nothing - all redeems need to be called manually by the
+        // strategist via redeemTickets
+        require(false, "use redeemTickets() instead");
+    }
+
+    /**
+     * @notice Redeem specific tickets from the Queue.
+     * Called by the strategist.
+     * @param _nftIds Array of NFT IDs to redeem
+     */
+    function redeemTickets(uint256[] memory _nftIds, uint256 expectedAmount)
```

Comment:
nit: Could probably add a `expectedAmount > 0` check

Reply:
Will keep as is on this one.